component/client/ClientCore.jsx

```diff
@@ -0,0 +1,53 @@
+import React, {Component} from "react";
+import {Grid, Row} from "react-bootstrap";
+import ClientNavbar from "./ClientNavbar.jsx";
+
+class ClientCore extends Component {
+  constructor(props) {
+    super(props);
+
+    // setup components map.
+    // TODO: add lifecycle to go properly set state based on persistant use.
+    // TODO: create chat and setting components.
+    this.components = {
+      "chat": <h1> chat </h1>,
+      "setting": <h1> setting </h1>
+    };
+
+    // initialize the activeKey.
+    this.state = {
+      activeKey: "chat"
+    };
+
+    // bind helper functions.
+    this.handleSelection = this.handleSelection.bind(this);
+  }
+
+  /*
+    Callback that handles Navbar transitions.
+  */
+  handleSelection(eventKey) {
```

Comment:
It is a component method. I defined it here so that it has access to the `this` binding of state. Namely, I needed to call the `setState` method to trigger rerendering if need be.